yoojin_lee/LV_1/예산.js

```diff
@@ -0,0 +1,16 @@
+function solution(d, budget) {
+    let answer = 0;
+    let sum = 0;
+    d.sort((a, b) => a - b);
+    // console.log(d)
+    for (let i = 0; i < d.length; i++) {
+        sum += d[i];
+        if (sum > budget) break;
+        answer += 1;
+    } return answer;
+}
```

Comment:
```suggestion
function solution(d, budget) {
    let answer = 0;
    const sortedD = [...d].sort((a, b) => a - b)

    for (let i in sortedD) {        
        budget -= sortedD[i]
        if (0 > budget) break;
        answer ++;
    }
    return answer;
}
```
해당 방식으로 풀이하면 원본 배열을 바꾸지 않을 수 있고, 문제에서 주어진 budget을 더 활용함으로써 sum이라는 변수를 하나 더 할당하지 않아도 됩니다.👀 reduce를 활용한 풀이도 올려 두었으니 참고하시면 좋을 것 같습니다! 잘 봤습니다!👍

Reply:
오!! reduce 활용해서 풀어보고 싶었지만  어떻게 풀어야할지 감이 안왔는데 참고하러 가보겠습니다!! 
코드 리뷰 감사합니다~!! 😄